### ISSUE TITLE
Styled Components Typescript type system fails with React.HTMLProps

### ISSUE THREAD

**Issue by jamesmfriedman** [styled-components/styled-components#3084]
## System:
 - OS: macOS 10.15.4
 - CPU: (12) x64 Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz
 - Memory: 501.05 MB / 32.00 GB
 - Shell: 5.7.1 - /bin/zsh
## Binaries:
 - Node: 11.14.0 - /usr/local/bin/node
 - npm: 6.13.4 - /usr/local/bin/npm
 - Watchman: 4.9.0 - /usr/local/bin/watchman
## npmPackages:
 - styled-components: ^5.0.1 => 5.0.1 

## Reproduction
https://codesandbox.io/s/quizzical-lichterman-0zhvx

I'm the maintainer of RMWC, and this is an issue that we can't seem to get rid of...

Original thread. Most relevant explanation at the bottom with screenshots from VSCode
https://github.com/jamesmfriedman/rmwc/issues/501#event-3184734226

Basically we can't make Style components type system happy anytime we extend React.HTMLProps...

**Comment by kitten** [comment]
Sorry, I'll have to close this.

We don't maintain any TypeScript typings in the main repository and are not updating the ones that live in `definitely-typed`, so we can't offer any help on this here. It's possible that if you browse past issues you'll find similar questions, but overall this repository's issues are not the right place to report these issues or to discuss them.
